Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -465,6 +569,82 @@ private extension RegistryManager {
     }
 }
 
+// MARK: - Serialization
+
+extension RegistryClient {
+    public enum Serialization {
```

Comment:
I am not sure expanding a few lines of JSON processing code into JSON models is necessary. I don't think we care about most of the contents. e.g., we only care if a `release` has `problem` and don't care what the actual problem is.

Reply:
I ended up with needing these for the test infrastructure which mimics the server, so instead of doing it only there I reused on both ends

Reply:
ok